.github/workflows/build.yml

```
@@ -1,10 +1,17 @@
+# PUBLIC — reusable build (compile + unit tests). Runs on push to main, on manual dispatch,
+# and is called by merge-gate.yml on pull requests. Never triggers directly on pull_request.
 name: Gradle Compile & UT
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    branches: [ main ]
```

YAMLlint (1.37.1)

[error] too many spaces inside brackets

(brackets)